ribs/emitters/_gradient_emitter.py

```diff
@@ -0,0 +1,297 @@
+            solution_dim=archive.solution_dim,
+            bounds=bounds,
+        )
+        self._initial_solutions = None
+        self._x0 = None
+
+        if x0 is None and initial_solutions is None:
+            raise ValueError("Either x0 or initial_solutions must be provided.")
+        if x0 is not None and initial_solutions is not None:
+            raise ValueError(
+                "x0 and initial_solutions cannot both be provided.")
+
+        if x0 is not None:
+            self._x0 = np.array(x0, dtype=archive.dtype)
+            check_1d_shape(self._x0, "x0", archive.solution_dim,
+                           "archive.solution_dim")
+        elif initial_solutions is not None:
+            self._initial_solutions = np.asarray(initial_solutions,
+                                                 dtype=archive.dtype)
+            check_batch_shape(self._initial_solutions, "initial_solutions",
+                              archive.solution_dim, "archive.solution_dim")
+
+        self._rng = np.random.default_rng(seed)
+        self._sigma0 = archive.dtype(sigma0) if isinstance(
+            sigma0, (float, np.floating)) else np.array(sigma0)
+        self._sigma_g = archive.dtype(sigma_g)
+        self._line_sigma = line_sigma
+        self._use_isolinedd = operator_type != 'isotropic'
+        self._measure_gradients = measure_gradients
+        self._normalize_grad = normalize_grad
+        self._epsilon = epsilon
+        self._batch_size = batch_size
+
+        self._jacobian_batch = None
+        self._parents = None
+
+    @property
+    def initial_solutions(self):
+        """numpy.ndarray: The initial solutions which are returned when the
+        archive is empty (if x0 is not set)."""
+        return self._initial_solutions
+
+    @property
+    def x0(self):
+        """numpy.ndarray: Center of the Gaussian distribution from which to
+        sample solutions when the archive is empty (if initial_solutions is not
+        set)."""
+        return self._x0
+
+    @property
+    def sigma0(self):
+        """float or numpy.ndarray: Standard deviation of the (diagonal) Gaussian
+        distribution."""
+        return self._sigma0
+
+    @property
+    def batch_size(self):
+        """int: Number of solutions to return in :meth:`ask`."""
+        return self._batch_size
+
+    @property
+    def batch_size_dqd(self):
+        """int: Number of solutions to return in :meth:`ask_dqd`."""
+        return self._batch_size
+
+    @property
+    def epsilon(self):
+        """int: The epsilon added for numerical stability when normalizing
+        gradients in :meth:`tell_dqd`."""
+        return self._epsilon
+
+    def ask_dqd(self):
+        """Create new solutions by sampling elites from the archive with 
+        (optional) Gaussian perturbation.
+
+        If the archive is empty and initial_solutions is given, the call to this 
+        method on the first iteration returns no solutions. Later iterations will 
```

Comment:
Take out the part about iterations. The if statements do not consider iterations.